src/preparers.py

```diff
@@ -864,3 +864,17 @@ def prepare_bactrianx(row):
         {"from": "user", "text": inputs, "parent": row["_source"]},
         {"from": "assistant", "text": outputs, "parent": 0},
     ]
+
+def prepare_mathdial(row):
+    conversation = row["conversation"].split("|EOM|")
+    parent = "mathdial"
+    messages = []
+    for i, turn in enumerate(conversation):
+        colon_index = turn.find(":")
+        messages.append({
+            "from": "user" if turn[:colon_index].strip() == "Teacher" else "assistant",
```

Comment:
If you think it makes more sense to switch user and assistant, so that the user is the student, then let's do that.